File: java/src/test/java/com/bc/libwally/CoreTest.java
package com.bc.libwally;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import static com.bc.libwally.core.Core.base582Bytes;
import static com.bc.libwally.core.Core.bytes2Base58;
import static com.bc.libwally.core.Core.bytes2Hex;
import static com.bc.libwally.core.Core.hex2Bytes;
import static org.junit.Assert.assertArrayEquals;
import static org.junit.Assert.assertEquals;

@RunWith(JUnit4.class)
public class CoreTest {

    @Test
    public void testBytes2Hex() {
        byte[] bytes = new byte[]{0x01,
                                  (byte) 0xAF,
                                  0x5F,
                                  (byte) 0xF2,
                                  (byte) 0x9A,
                                  0x12,
                                  (byte) 0x89,
                                  (byte) 0xAA,
                                  (byte) 0xBA,
                                  0x45,
                                  0x00};

        String expectedHex = "01af5ff29a1289aaba4500";
        assertEquals(expectedHex, bytes2Hex(bytes));
    }

    @Test
    public void testHex2Bytes() {
        String hex = "01af5ff29a1289aaba4500";
        byte[] expectedBytes = new byte[]{0x01,
                                          (byte) 0xAF,
                                          0x5F,
                                          (byte) 0xF2,
                                          (byte) 0x9A,
                                          0x12,
                                          (byte) 0x89,
                                          (byte) 0xAA,
                                          (byte) 0xBA,
                                          0x45,
                                          0x00};
        assertArrayEquals(expectedBytes, hex2Bytes(hex));
    }

    @Test
    public void testBytes2Base58() {
        byte[] bytes = new byte[]{0x01,
                                  (byte) 0xAF,
                                  0x5F,
                                  (byte) 0xF2,
                                  (byte) 0x9A,
                                  0x12,
                                  (byte) 0x89,
                                  (byte) 0xAA,
                                  (byte) 0xBA,
                                  0x45,
                                  0x00};
        String expectedBase58 = "3jb7AeBHgv74mmN2YbA2";
        assertEquals(expectedBase58, bytes2Base58(bytes));
    }

    @Test
    public void testBase582Bytes() {
        String base58 = "3jb7AeBHgv74mmN2YbA2";
        byte[] expectedBytes = new byte[]{0x01,
                                          (byte) 0xAF,
                                          0x5F,
                                          (byte) 0xF2,
                                          (byte) 0x9A,
                                          0x12,
                                          (byte) 0x89,
                                          (byte) 0xAA,
                                          (byte) 0xBA,
                                          0x45,
                                          0x00};
        assertArrayEquals(expectedBytes, base582Bytes(base58));
    }

}

File: android/app/src/androidTest/java/com/bc/libwally/CoreTest.kt
package com.bc.libwally

import androidx.test.ext.junit.runners.AndroidJUnit4
import com.bc.libwally.core.Core.*
import org.junit.Assert.assertArrayEquals
import org.junit.Assert.assertEquals
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class CoreTest {

    @Test
    fun testBytes2Hex() {
        val bytes = byteArrayOf(
            0x01,
            0xAF.toByte(),
            0x5F,
            0xF2.toByte(),
            0x9A.toByte(),
            0x12,
            0x89.toByte(),
            0xAA.toByte(),
            0xBA.toByte(),
            0x45,
            0x00
        )
        val expectedHex = "01af5ff29a1289aaba4500"
        assertEquals(expectedHex, bytes2Hex(bytes))
    }

    @Test
    fun testHex2Bytes() {
        val hex = "01af5ff29a1289aaba4500"
        val expectedBytes = byteArrayOf(
            0x01,
            0xAF.toByte(),
            0x5F,
            0xF2.toByte(),
            0x9A.toByte(),
            0x12,
            0x89.toByte(),
            0xAA.toByte(),
            0xBA.toByte(),
            0x45,
            0x00
        )
        assertArrayEquals(expectedBytes, hex2Bytes(hex))
    }

    @Test
    fun testBytes2Base58() {
        val bytes = byteArrayOf(
            0x01,
            0xAF.toByte(),
            0x5F,
            0xF2.toByte(),
            0x9A.toByte(),
            0x12,
            0x89.toByte(),
            0xAA.toByte(),
            0xBA.toByte(),
            0x45,
            0x00
        )
        val expectedBase58 = "3jb7AeBHgv74mmN2YbA2"
        assertEquals(expectedBase58, bytes2Base58(bytes))
    }

    @Test
    fun testBase582Bytes() {
        val base58 = "3jb7AeBHgv74mmN2YbA2"
        val expectedBytes = byteArrayOf(
            0x01,
            0xAF.toByte(),
            0x5F,
            0xF2.toByte(),
            0x9A.toByte(),
            0x12,
            0x89.toByte(),
            0xAA.toByte(),
            0xBA.toByte(),
            0x45,
            0x00
        )
        assertArrayEquals(expectedBytes, base582Bytes(base58))
    }
}